operator/v1/tests/clustercsidrivers.operator.openshift.io/VSphereConfigurableMaxAllowedBlockVolumesPerNode.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "ClusterCSIDriver"
+crdName: clustercsidrivers.operator.openshift.io
+featureGates:
+  - VSphereConfigurableMaxAllowedBlockVolumesPerNode
+tests:
+  onCreate:
+    - name: Should be able to create ClusterCSIDriver with volume limit option
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: ClusterCSIDriver
+        metadata:
+          name: csi.sharedresource.openshift.io
+        spec:
+          driverConfig:
+            driverType: vSphere
+            vSphere:
+              maxAllowedBlockVolumesPerNode: 255
+          logLevel: Normal
+          operatorLogLevel: Normal
+      expected: |
+        apiVersion: operator.openshift.io/v1
+        kind: ClusterCSIDriver
+        metadata:
+          name: csi.sharedresource.openshift.io
+        spec:
+          driverConfig:
+            driverType: vSphere
+          vSphere:
+              maxAllowedBlockVolumesPerNode: 255
+          logLevel: Normal
+          operatorLogLevel: Normal
+    - name: Should be able to create a minimal ClusterCSIDriver
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: ClusterCSIDriver
+        metadata:
+          name: csi.sharedresource.openshift.io
+        spec: {} # No spec is required for a ClusterCSIDriver
+      expected: |
+        apiVersion: operator.openshift.io/v1
+        kind: ClusterCSIDriver
+        metadata:
+          name: csi.sharedresource.openshift.io
+        spec:
+          logLevel: Normal
+          operatorLogLevel: Normal
+    - name: Should not be able to set limit above allowed range
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: ClusterCSIDriver
+        metadata:
+          name: csi.sharedresource.openshift.io
+        spec:
+          driverConfig:
+            driverType: vSphere
+            vSphere:
+              maxAllowedBlockVolumesPerNode: 256
+          logLevel: Normal
+          operatorLogLevel: Normal
+      expectedError: "Invalid value: 256: spec.driverConfig.vSphere.maxAllowedBlockVolumesPerNode in body should be less than or equal to 255"
+    - name: Should not be able to set limit below allowed range
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: ClusterCSIDriver
+        metadata:
+          name: csi.sharedresource.openshift.io
+        spec:
+          driverConfig:
+            driverType: vSphere
+            vSphere:
+              maxAllowedBlockVolumesPerNode: -1
+          logLevel: Normal
+          operatorLogLevel: Normal
+      expectedError: "Invalid value: -1: spec.driverConfig.vSphere.maxAllowedBlockVolumesPerNode in body should be greater than or equal to 0"
```

Comment:
Why allow `0` value at all btw?

Reply:
@gnufied The only reason I though it should be allowed is because it gives users the option to disable the limit. I really can't think if a good practical use case apart from some experimental scenarios.

I can also imagine the limit increasing in the future, and if we don't reflect that in the API fast enough it might help to have this option just to unblock users temporarily - although this is not too safe. I think not allowing `0` here is also a valid choice, no strong opinions on this. Do you want me to change the range to 1-255?

Reply:
Yes, current limit is 59 and we haven't had too many complaints. The thing is - driver already guards against >255 limits, but there is no safeguard if user sets the limit to `0` - https://github.com/openshift/vmware-vsphere-csi-driver/blob/master/pkg/csi/service/node.go#L429 and hence I would stop users from setting the limit to 0.

We can always relax this validation in future. It is always useful to start with more restricted baseline and relax afterwards than the other way around. If we allowed `0` today and then we found, users are breaking themselves, we won't be able to roll it back later.

Reply:
Fair point - I've adjusted the range to 1-255.